skiko/src/awtTest/kotlin/org/jetbrains/skiko/SkiaSystemFontProviderTest.kt

```diff
@@ -0,0 +1,69 @@
+package org.jetbrains.skiko
+
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.runTest
+import org.junit.Assume
+import org.junit.Test
+import kotlin.test.*
+
+
+@OptIn(ExperimentalCoroutinesApi::class)
+class SkiaSystemFontProviderTest {
+    private val provider = SystemFontProvider.skia
+
+    @Test
+    fun `should contain at least some font families`() = runTest {
+        val families = provider.familyNames()
+        assertTrue(families.isNotEmpty(), "Available font families must not be empty")
+    }
+
+    @Test
+    fun `should include 'System Font' and its AWT alias in system font families (only on macOS)`() = runTest {
+        Assume.assumeTrue(hostOs == OS.MacOS)
+
+        val systemFamilies = provider.familyNames()
+        assertContains(
+            iterable = systemFamilies,
+            element = FontFamilyKey.Apple.AppleSystemUiFont.familyName,
+            message = ".AppleSystemUIFont not found in system families"
+        )
+        assertContains(
+            iterable = systemFamilies,
+            element = FontFamilyKey.Apple.SystemFont.familyName,
+            message = "System Font not found in system families"
+        )
+    }
+
+    @Test
+    fun `should provide all the system fonts also available via AWT`() = runTest {
```

Comment:
Ubuntu:
```
java.lang.AssertionError: These AWT font families are missing:
 * Droid Sans
 * Droid Sans Mono
 * Droid Sans Mono Dotted
 * Droid Sans Mono Slashed
 * Droid Serif
 * Fira Code
 * Inconsolata
 * Inter
 * JetBrains Mono
 * Roboto
 * Source Code Pro
```
They are also in the JBR folder

Reply:
MacOs Monterey 12.5., JBR17:
```
0 = ".SF NS"
1 = ".SF NS Mono"
2 = "Athelas"
3 = "Courier"
4 = "Droid Sans"
5 = "Droid Sans Mono"
6 = "Droid Sans Mono Dotted"
7 = "Droid Sans Mono Slashed"
8 = "Droid Serif"
9 = "Fira Code"
10 = "Hiragino Kaku Gothic Pro"
...
15 = "Hiragino Mincho Pro"
16 = "Inconsolata"
17 = "Inter"
18 = "Iowan Old Style"
19 = "JetBrains Mono"
20 = "Marion"
21 = "Noto Sans Adlam"
...
122 = "Noto Serif Yezidi"
123 = "Roboto Light"
124 = "Roboto Thin"
125 = "Seravek"
126 = "Source Code Pro"
127 = "Superclarendon"
128 = "Times"
```
Those are in the JBR folder, in /System/Library/Fonts/Supplementa, and in /System/Library/Fonts/Supplemental.

Don't know what happens, but usual mac applications (Pages, TextEdit, Chrome) also don't see fonts from /System/Library/Fonts.

Reply:
Hmmm I haven't run the tests on JBR I suppose, but I can look into this. JDK classpath fonts are not a big deal, as this seems another JBR-only issue (the tests pass for me when not using JBR, and on the CI), since JBR embeds some fonts. I'll look into loading those fonts into the list too, but there is a concern, in the sense that non-system fonts aren't accessible to Skia automatically. What I was thinking is adding a "if you see fonts in the JDK, add them to the list of custom fonts so they can be used normally" — they are, after all, not system fonts.

As for the reverse, Skia having fonts that AWT doesn't know about, that's ignored by design. AWT can't deal with a number of font types (variable fonts, composite fonts, COLRv1 emoji fonts, etc) since its implementation is stuck to 20+ years ago, so it's only normal Skia knows about more fonts than AWT. For the AWT -> Skia interop this isn't a problem as long as we guarantee all AWT fonts are available to Skia (once the JDK classpath fonts loading is added). For the Skia -> AWT interop there isn't much we can do, but users can either rely on AWT's silent font substitution mechanism, or manually check the available AWT font families and decide what to do if missing.

Reply:
We still have this issue. What is the decision to resolve it?

We have fonts in the JDK folder, and have some fonts `/System/Library` on macOS. How do we register them? If there is a way to load them, we need to use it in tests, so they don't fail. If there is no way, we need to find another solution (maybe even fillter fonts in the IDEA settings itself)